src/App.tsx

```diff
@@ -17,18 +73,31 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                inputQuery={inputQuery}
+                onFilterInput={onFilterInput}
+                onFilterSelect={onFilterSelect}
+                clearFilterInput={clearFilterInput}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading && todos.length > 0 && (
+                <TodoList
+                  todos={todos}
+                  selectedTodoId={selectedTodo ? selectedTodo.id : 0}
```

Comment:
The `selectedTodoId` prop defaults to `0`, which could be a valid ID. Consider using `null` or another invalid ID as a default to prevent potential issues.